polyfill/test262-harness.mjs

```diff
@@ -0,0 +1,246 @@
+import color from 'ansi-colors';
+import fs from 'node:fs';
+import globSync from 'tiny-glob/sync.js';
+import path from 'node:path';
+import process from 'node:process';
+import ProgressBar from 'progress';
+import util from 'node:util';
+import vm from 'node:vm';
+import yaml from 'js-yaml';
+
+// = Temporal test262 runner =
+
+// This is a lightweight runner for Temporal's test262 tests. It exists in order
+// to be fast, and it is specific to Temporal. It doesn't provide all of the
+// facilities that the full test262 suite would expect. It achieves the "fast"
+// goal by parsing the Temporal polyfill once into a vm.Script, instead of
+// reading and parsing the whole file once for each test like the prelude option
+// of test262-harness would.
+
+// Without any arguments, all of the Temporal tests will be run. The script
+// takes one argument: a glob pattern relative to Temporal/ that specifies a
+// subset of tests to be run. For example:
+//   "PlainDateTime/**"
+//   "*/prototype/with/*.js"
+//   "**/basic.js"
```

Comment:
Questions:
* To which Temporal/ is the pattern relative? ~~Is it the test/built-ins/Temporal/ directory of a test262 repository, e.g. https://github.com/tc39/test262/tree/d45766bef3519eb9da6d1957082bd0b026949e90/test/built-ins/Temporal ?~~ EDIT: I see below that it is relative to _any_ Temporal directory that is a grandchild of test262/test/.
  ```suggestion
  // takes one argument: a glob pattern relative to test262/test/*/Temporal/ that specifies a
  // subset of tests to be run. For example:
  //   "PlainDateTime/**"
  //   "*/prototype/with/*.js"
  //   "**/basic.js"
  ```
  * And if so, does that mean it cannot run tests in staging?
* Why is the pattern relative to that directory rather than to the working directory, which would support tab completion? I could see having a special form for the special root (e.g., a `--relative` flag or even arguments like `'$TEST262_TEMPORAL/PlainDateTime/**'` or possibly `"///PlainDateTime/**"`), but making it the default is strange.
* Why limit invocation to specifying a single pattern rather than an arbitrary number?

For reference, I just made changes to a similar test262.js in engine262 (although I couldn't change the relative base behavior without breaking backwards compatibility, so I instead added working directory as a fallback base rather than the default): https://github.com/engine262/engine262/commit/9e870f674c629a9e610966308c1df1a066a9e979#diff-c234e9bf4f04d8032e409fafad31006e78f09a0cf62a411dbc05ceb8478778b3R53-R64

Reply:
I did it this way for a couple of reasons: mostly, so that users of the script don't have to remember that there could be Temporal tests matching the pattern in any one of `built-ins/`, `intl402/`, and `staging/`. (If you unknowingly use tab completion to help you write e.g. `test262/test/built-ins/Temporal/TimeZone/**`, you'll miss the `intl402/` and `staging/` tests.)

A secondary reason is that a pattern like `**/length.js` would run all ~600 of test262's `length.js` tests instead of just the ~200 ones relevant to Temporal.

I like the idea of supporting both, with working directory as a fallback.

I hadn't allowed multiple glob patterns because I hadn't needed it yet, but it seems simple enough to add, and useful.